app/src/main/java/org/kinecosystem/kinit/view/TabsAdapter.kt

```diff
@@ -27,7 +28,7 @@ import javax.inject.Inject
 private const val NUMBER_OF_TABS = 5
 private const val PRE_EARN_SHOW_DURATION: Long = 2000L
 
-class TabsAdapter :
+class TabsAdapter(val mainActivity: MainActivity) :
```

Comment:
It's bad practice to pass the Activity instance into the adapter, you can create an interface to invoke the `enblePageSelection()` method,